R/utils.R

```diff
@@ -349,6 +349,38 @@
     altExps(x) <- values
     return(x)
 }
+
+# This function can be used to add values to reducedDims
+.add_values_to_reducedDims <- function(x, values, name, ...){
+    # Check values
+    if( !((is(values, "matrix") && length(values) > 0)) ){
+        stop("'values' must be a matrix.", call. = FALSE)
+    }
+    # Check names
+    if( !.is_a_string(name) ){
+        stop("'name' must be a character value.", call. = FALSE)
+    }
+    #
+    # If the object is SE, convert it to TreeSE
+    if( !is(x, "SingleCellExperiment") ){
+        x <- as(x, "TreeSummarizedExperiment")
+        warning(
+            "SummarizedExperiment does not have reducedDims slot. ",
+            "Therefore, it is converted to TreeSummarizedExperiment.",
+            call. = FALSE)
+    }
+    # Throw warning if values of reducedDim are overwritten
+    if ( name %in% names(reducedDims(x)) ){
+        warning(
+          "The following values are already present in `reducedDims` and will ",
+          "be overwritten: '",
+          name,
+          "'. Consider using the 'name' argument to specify alternative ",
+          "names.", call. = FALSE)
+    }
```

Comment:
Indentation is off

Reply:
I fixed the indentation and modified the input check to allow dist objects as well as matrices.